### PR TITLE
[CPDLP-3493] NPQ Internal API - Add feature flag instead of application config

### PR DESCRIPTION
**Pull request by mooktakim**
### Context

Ticket: https://dfedigital.atlassian.net.mcas.ms/browse/CPDLP-3493

### Changes proposed in this pull request

- Added feature: `Feature.ecf_api_disabled?` with flipper

### CODE REVIEW

File: app/services/feature.rb [hunk 1]
```diff
@@ -51,5 +54,9 @@ def disable_registration!
     def enable_registration!
       Flipper.disable(REGISTRATION_DISABLED)
     end
+
+    def ecf_api_disabled?
```
Comment (leandroalemao) [inline app/services/feature.rb:58]: Hi @mooktakim Would be nice to add a spec for this method?

**Review by leandroalemao — COMMENTED** [review]
Also would be nice to add a method to enable the feature flag? ie: `disable_ecf_api!`

**Review by ethax-ross — COMMENTED** [review]
Looks good past Leandro's comment

**Comment by cwrw** [comment]
Looks good to me too, @leandroalemao regarding toggling I was wondering if we do need toggling methods as those are given to us by `Flipper` so need to add a wrapper method?

**Comment by mooktakim** [comment]
> Looks good to me too, @leandroalemao regarding toggling I was wondering if we do need toggling methods as those are given to us by `Flipper` so need to add a wrapper method?

I believe flipper has an admin UI to enable/disable features, unlikely to need the toggle:
`/admin/feature_flags/features`

File: spec/services/feature_spec.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+require "rails_helper"
+
+RSpec.describe Feature do
```
Comment (ethax-ross) [inline spec/services/feature_spec.rb:3]: There's already a spec for this; can we consolidate them? https://github.com/DFE-Digital/npq-registration/blob/main/spec/lib/services/feature_spec.rb

Comment (ethax-ross) [reply spec/services/feature_spec.rb:3]: might be worth improving the general coverage of this service while you're in there; looks like only the `registration_closed?` method is tested at the moment

Comment (mooktakim) [reply spec/services/feature_spec.rb:3]: 🤦 i will merge it into the correct location.

**Review by cwrw — APPROVED** [review]

**Review by ethax-ross — APPROVED** [review]

**Pull request merged**